### PR TITLE
chore(editor3): Add unit tests for style suggestions

### PR DESCRIPTION
**Pull request by ioanpocol**

### CODE REVIEW

File: scripts/core/editor3/reducers/tests/suggestion_setup.jsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    return newRawContent;
+}
+
+export function getInitialEditorState(rawContent) {
+    const content = convertFromRaw(prepareRawContent(rawContent));
+
+    return EditorState.createWithContent(content);
+}
+
+export function applySelection(editorState, startBlockIndex, startOffset, endBlockIndex, endOffset) {
```
Comment (tomaskikutis) [inline scripts/core/editor3/reducers/tests/suggestion_setup.jsx]: This seems simpler to me
https://github.com/superdesk/superdesk-client-core/blob/687d4259f6bbc619d62a6961d79f744308c4cd88/scripts/core/editor3/components/tests/multipleHighlights.spec.jsx#L78

File: scripts/core/editor3/reducers/tests/suggestion_style.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+        expect(Highlights.getHighlightData(editorState, 'TOGGLE_BOLD_SUGGESTION-1')).toEqual({
+            date: date,
+            author: 'author_id',
+            type: 'TOGGLE_BOLD_SUGGESTION',
+            originalStyle: '',
+        });
+    });
+
+    it('CREATE_CHANGE_STYLE_SUGGESTION - set BOLD suggestion for a bold text', () => {
```
Comment (tomaskikutis) [inline scripts/core/editor3/reducers/tests/suggestion_style.spec.jsx]: When creating a test case, describe the expected outcome. Write the "it" case so it's gramatically correct when you read it including "it".

`it('Should set BOLD suggestion even if the text is already bold')` or
`it('Should ignore BOLD suggestion if the text is already bold')`

The same applies to other tests like "set BOLD suggestion 2 times" where it's not specified what is the expected behaviour after setting it 2 times.

File: scripts/core/editor3/reducers/tests/suggestion_setup.jsx (hunk 1 above)
Comment (ioanpocol) [reply scripts/core/editor3/reducers/tests/suggestion_setup.jsx]: I think the current solution is fine, it cover also the selection of all blocks.

File: scripts/core/editor3/reducers/tests/suggestion_style.spec.jsx [hunk 3]
```diff
@@ -53,7 +53,7 @@ describe('editor3.reducers.suggestion.CREATE_CHANGE_STYLE_SUGGESTION', () => {
         });
     });
 
-    it('CREATE_CHANGE_STYLE_SUGGESTION - set BOLD suggestion for a bold text', () => {
+    it('should set BOLD suggestion for a bold text, the BOLD style not present anymore for selected text', () => {
```
Comment (tomaskikutis) [inline scripts/core/editor3/reducers/tests/suggestion_style.spec.jsx]: Sorry for picking on the details, but I think it important to have correct and properly structured sentences

* it should create a `BOLD` suggestion
* it should create a `REMOVE-BOLD` suggestion when toggling `BOLD` on text which was already bold before enabling suggestion mode
* it should not add any suggestions after toggling 'BOLD' twice
* ...

File: scripts/core/editor3/reducers/tests/suggestion_style.spec.jsx [hunk 4]
```diff
@@ -141,7 +141,7 @@ describe('editor3.reducers.suggestion.CREATE_CHANGE_STYLE_SUGGESTION', () => {
         }
     });
 
-    it('CREATE_CHANGE_STYLE_SUGGESTION - set BOLD partially overlapping suggestions', () => {
+    it('should overwrite overlapped section when set 2 BOLD overlapped suggestions', () => {
```
Comment (tomaskikutis) [inline scripts/core/editor3/reducers/tests/suggestion_style.spec.jsx]: Clarify the description, I can't understand what does it do from the description alone

File: scripts/core/editor3/reducers/tests/suggestion_style.spec.jsx (hunk 3 above)
Comment (ioanpocol) [reply scripts/core/editor3/reducers/tests/suggestion_style.spec.jsx]: Fix is in the next PR

File: scripts/core/editor3/reducers/tests/suggestion_style.spec.jsx (hunk 4 above)
Comment (ioanpocol) [reply scripts/core/editor3/reducers/tests/suggestion_style.spec.jsx]: fix is in the next PR

**Pull request merged**